### PR TITLE
LGTM: Cleanup and retarget to .Net 4.6.1

### LINKED ISSUE #2: Use NuGet to reference iTextSharp
So that we can remove another binary in
e-TaxInvoice-PDFgen/eTaxInvoicePdfGenerator/libsITEXT/

https://www.nuget.org/packages/iTextSharp/

### PR DESCRIPTION
**Pull request by mahasak**
- Remove unnecessary binary from repository and make it nuget reference
- Change target framework to .Net 4.6.1 to prepared for .Net Standard binary
- Fix issue to compile its on clone from master
- Add dependency list to README.md

- Close #2 #9 #10

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,4 @@ eTaxInvoicePdfGenerator/bin/
 /eTaxInvoicePdfGenerator.csproj.user
 eTaxInvoicePdfGenerator/obj/
 eTaxInvoicePdfGenerator/publish/
+packages/
```
Comment (mahasak) [inline .gitignore:6]: Add nuget package temporary folder to .gitignore. So we're not going to add unnecessary binary to repository.

File: README.md [hunk 2]
```diff
@@ -34,12 +34,14 @@ Windows x64 : [โปรแกรมจัดเตรียมใบกำก
...
- * itextsharp.xtra.dll
-
+## Dependency
+ * itextsharp (version 5.5.11)
+ * itextsharp.pdfa (version 5.5.11)
+ * itextsharp.xtra (version 5.5.11)
+ * System.Data.SQLite.Core (version 1.0.105.2+)
+ * Microsoft.ReportViewer.Common (version 10.0.40219.1+)
+ * Microsoft.ReportViewer.WinForms (version 10.0.40219.1+)
+
```
Comment (mahasak) [inline README.md:44]: Update dependency list, make it correct

File: eTaxInvoicePdfGenerator/eTaxInvoicePdfGenerator.csproj [hunk 3]
```diff
@@ -109,26 +116,36 @@
...
-    <Reference Include="itextsharp.pdfa">
-      <HintPath>libsITEXT\itextsharp.pdfa.dll</HintPath>
+    <Reference Include="itextsharp.pdfa, Version=5.5.11.0, Culture=neutral, PublicKeyToken=8354ae6d2174ddca, processorArchitecture=MSIL">
+      <HintPath>..\packages\itextsharp.pdfa.5.5.11\lib\itextsharp.pdfa.dll</HintPath>
     </Reference>
-    <Reference Include="itextsharp.xtra">
-      <HintPath>libsITEXT\itextsharp.xtra.dll</HintPath>
+    <Reference Include="itextsharp.xtra, Version=5.5.11.0, Culture=neutral, PublicKeyToken=8354ae6d2174ddca, processorArchitecture=MSIL">
+      <HintPath>..\packages\itextsharp.xtra.5.5.11\lib\itextsharp.xtra.dll</HintPath>
+    </Reference>
```
Comment (mahasak) [inline eTaxInvoicePdfGenerator/eTaxInvoicePdfGenerator.csproj:127]: Change these references from local DLL to nuget references

File: eTaxInvoicePdfGenerator/eTaxInvoicePdfGenerator.csproj [hunk 4]
```diff
@@ -109,26 +116,36 @@
...
+    <Reference Include="Microsoft.ReportViewer.DataVisualization, Version=10.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
+      <HintPath>..\packages\Microsoft.ReportViewer.Common.10.0.40219.1\lib\Microsoft.ReportViewer.DataVisualization.dll</HintPath>
+    </Reference>
+    <Reference Include="Microsoft.ReportViewer.ProcessingObjectModel, Version=10.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
+      <HintPath>..\packages\Microsoft.ReportViewer.Common.10.0.40219.1\lib\Microsoft.ReportViewer.ProcessingObjectModel.dll</HintPath>
+    </Reference>
+    <Reference Include="Microsoft.ReportViewer.WinForms, Version=10.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
+      <HintPath>..\packages\Microsoft.ReportViewer.WinForms.10.0.40219.1\lib\Microsoft.ReportViewer.WinForms.dll</HintPath>
     </Reference>
```
Comment (mahasak) [inline eTaxInvoicePdfGenerator/eTaxInvoicePdfGenerator.csproj:139]: Add necessary reporting library from nuget references

File: eTaxInvoicePdfGenerator/eTaxInvoicePdfGenerator.csproj [hunk 5]
```diff
@@ -109,26 +116,36 @@
...
     <Reference Include="System" />
     <Reference Include="System.configuration" />
     <Reference Include="System.Data" />
-    <Reference Include="System.Data.SQLite, Version=1.0.101.0, Culture=neutral, PublicKeyToken=db937bc2d44ff139, processorArchitecture=AMD64">
-      <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\..\Lib\sqlite-netFx40-binary-bundle-x64-2010-1.0.101.0\System.Data.SQLite.dll</HintPath>
+    <Reference Include="System.Data.SQLite, Version=1.0.105.2, Culture=neutral, PublicKeyToken=db937bc2d44ff139, processorArchitecture=MSIL">
+      <HintPath>..\packages\System.Data.SQLite.Core.1.0.105.2\lib\net46\System.Data.SQLite.dll</HintPath>
     </Reference>
```
Comment (mahasak) [inline eTaxInvoicePdfGenerator/eTaxInvoicePdfGenerator.csproj:149]: same as above

File: eTaxInvoicePdfGenerator/eTaxInvoicePdfGenerator.csproj [hunk 6]
```diff
@@ -571,59 +589,11 @@
...
-      <Visible>False</Visible>
-      <Group>
-      </Group>
-      <TargetPath>
-      </TargetPath>
-      <PublishState>Include</PublishState>
-      <IncludeHash>True</IncludeHash>
-      <FileType>Assembly</FileType>
-    </PublishFile>
     <PublishFile Include="Microsoft.ReportViewer.WinForms">
```
Comment (mahasak) [inline eTaxInvoicePdfGenerator/eTaxInvoicePdfGenerator.csproj:597]: see above

File: eTaxInvoicePdfGenerator/eTaxInvoicePdfGenerator.csproj [hunk 7]
```diff
@@ -760,11 +730,16 @@
     </COMReference>
   </ItemGroup>
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
-  <Import Project="$(SQLDBExtensionsRefPath)\Microsoft.Data.Tools.Schema.Sql.UnitTesting.targets" Condition="'$(SQLDBExtensionsRefPath)' != ''" />
-  <Import Project="$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)\SSDT\Microsoft.Data.Tools.Schema.Sql.UnitTesting.targets" Condition="'$(SQLDBExtensionsRefPath)' == ''" />
   <PropertyGroup>
```
Comment (mahasak) [inline eTaxInvoicePdfGenerator/eTaxInvoicePdfGenerator.csproj:733]: remove unused target from project, it create annoying warning.

File: eTaxInvoicePdfGenerator/eTaxInvoicePdfGenerator.csproj [hunk 8]
```diff
@@ -760,11 +730,16 @@
...
     <SsdtUnitTestVersion>2.0</SsdtUnitTestVersion>
   </PropertyGroup>
+  <Import Project="..\packages\System.Data.SQLite.Core.1.0.105.2\build\net46\System.Data.SQLite.Core.targets" Condition="Exists('..\packages\System.Data.SQLite.Core.1.0.105.2\build\net46\System.Data.SQLite.Core.targets')" />
+  <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
+    <PropertyGroup>
+      <ErrorText>This project references NuGet package(s) that are missing on this computer. Use NuGet Package Restore to download them.  For more information, see http://go.microsoft.com/fwlink/?LinkID=322105. The missing file is {0}.</ErrorText>
+    </PropertyGroup>
+    <Error Condition="!Exists('..\packages\System.Data.SQLite.Core.1.0.105.2\build\net46\System.Data.SQLite.Core.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\System.Data.SQLite.Core.1.0.105.2\build\net46\System.Data.SQLite.Core.targets'))" />
+  </Target>
```
Comment (mahasak) [inline eTaxInvoicePdfGenerator/eTaxInvoicePdfGenerator.csproj:742]: Add SQLite support without EF support as we not use it

File: eTaxInvoicePdfGenerator/packages.config [hunk 9]
```diff
@@ -0,0 +1,9 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="iTextSharp" version="5.5.11" targetFramework="net461" />
+  <package id="itextsharp.pdfa" version="5.5.11" targetFramework="net461" />
+  <package id="itextsharp.xtra" version="5.5.11" targetFramework="net461" />
+  <package id="Microsoft.ReportViewer.Common" version="10.0.40219.1" targetFramework="net461" />
+  <package id="Microsoft.ReportViewer.WinForms" version="10.0.40219.1" targetFramework="net461" />
+  <package id="System.Data.SQLite.Core" version="1.0.105.2" targetFramework="net461" />
+</packages>
```
Comment (mahasak) [inline eTaxInvoicePdfGenerator/packages.config:9]: Add nuget config to restore dependency on first time clone.

**Review by mahasak — COMMENTED** [review]
Please review these change amd LGTM

File: README.md (hunk 2 above)
Comment (varokas) [reply README.md]: We should make it clear that this is the dependencies included in the project (as NuGet or checked in to Git). So People won't confused whether they need to download these.

**Comment by varokas** [comment]
LGTM. See comment for questions. I'll leave this on for 1 day for more review.

**Comment by pragasitcha** [comment]
LGTM  Krub !!

**Pull request merged**